docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react'
+import { Menu, Popup, toolbarBehavior, popupFocusTrapBehavior } from '@stardust-ui/react'
+
+export interface PopoverProps {
+  className?: string
+}
+
+class Popover extends React.Component<PopoverProps> {
+  render() {
+    return (
+      <Menu
+        styles={{
+          transition: 'opacity 0.2s',
+          position: 'absolute',
+          top: '-20px',
+          right: '5px',
+          background: '#fff',
+          boxShadow: '0px 2px 4px #ddd',
+          borderRadius: '.3rem',
+          opacity: 0,
+
+          '& .smile-emoji': {
+            display: 'none',
+          },
+
+          '&.focused .smile-emoji': {
+            display: 'flex',
+          },
+
+          '&:hover .smile-emoji': {
+            display: 'flex',
+          },
+
+          '& a:focus': {
+            textDecoration: 'none',
+            color: 'inherit',
+          },
+
+          '& a': {
+            color: 'inherit',
+          },
+        }}
+        iconOnly
+        className={this.props.className}
+        items={[
+          { key: 'smile', icon: 'smile', className: 'smile-emoji' },
+          { key: 'smile2', icon: 'smile', className: 'smile-emoji' },
+          { key: 'smile3', icon: 'smile', className: 'smile-emoji' },
+          { key: 'a', icon: 'thumbs up' },
+          { key: 'c', icon: 'ellipsis horizontal' },
+        ]}
+        renderItem={renderContextMenu}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        accessibility={toolbarBehavior}
+        data-is-focusable={true}
+      />
+    )
+  }
+}
+
+export default Popover
+
+const renderContextMenu = (MenuItem, props) => {
+  if (props.icon !== 'ellipsis horizontal') {
+    return <MenuItem {...props} />
+  }
+
+  return (
+    <Popup
+      key={props.key}
+      position="below"
+      accessibility={popupFocusTrapBehavior}
+      trigger={<MenuItem {...props} />}
+      content={
+        <div
```

Comment:
@sophieH29 needs to use a `div` primitive and inline `style` in order to wrap a `Menu` for the content of a `Popup`; from our perspective, **are we ok with that?**

Shouldn't our users have a way to custom one-off things like this with a (surpris,e surprise) custom component that they can style and will go through the fella renderer too in order to provide all Stardust capabilities (accessibility, rtl, computed styles, etc)?

I'm thinking `createComponent` should be used here, but how? I was trying to prototype this and doesn't look smooth at all...

This example should become something like:

```jsx
const MyMenuWithWrapper = createComponent({
  displayName: 'MenuWrapper',

  render({ stardust }) {
    const { classes, rtl } = stardust // how do we pass style object to custom component?
    return <div className={classes}><Menu .../></div> // we still use `div`?
  },
})
```
and then use `MyMenuWithWrapper` for `Popup` `content`

can you guys weigh in here?

@layershifter @kuzhelov @mnajdova @miroslavstastny

Reply:
If you add in the props for the `MyMenuWithWrapper` styles and variables, those should work exactly the same way as they work for any first-class Stardust component. You can target MenuWrapper in the provider (by the `displayName`) or use `styles={{/* your custom styles*/}}` and `variables={{/* your custom variables */}}` on component level. BTW I would recommend using the same name for the Component name and the `displayName` arg to `createComponent`. This is the PR that contains examples for this: https://github.com/stardust-ui/react/pull/517 and is already merged, so on master you should be able to take a look into the Integrate Custom components page. Let me know if you need more input on this.

Reply:
@mnajdova, I see that there is a way to override it using ```createComponent```.
I was making this prototype actually as a user of Stardust, I combined all the needed components and received the result which I expected. 
So as a user, I have no clue why should I override it using ```createComponent``` :) 
And I see the rest of the prototypes are made in the same manner (without ```createComponent```), I am kind of confused. I read in your responses "what" and "how" to do it, but I didn't read "why"

Reply:
frankly, this was a proper "why" answer for me _These are not that valid for this specific usecase but prototypes should be our suggestions for users to design their code_  :)
why the ```createComponent``` exists I understand, the thing was about the reason it should be applied for this specific use case.
Anyway thank you for your responses, guys, really appreciate that.
I have a few more things I want to ask/clarify/ and maybe need your assistance. But probably let's discuss it tomorrow offline, as don't wanna take more of your time on replying.
Again, thank you!